tedana/decomposition/eigendecomp.py

```diff
@@ -216,14 +214,36 @@ def tedica(n_components, dd, conv, fixed_seed, cost, final_cost):
     Uses `mdp` implementation of FastICA for decomposition
     """
 
-    import mdp
+    from sklearn.decomposition import FastICA
+
+    def cost_func(cost):
+        """
+        Cost function supplied to ICA.
+
+        Parameters
+        ----------
+        cost:  {'tanh', 'pow3', 'gaus', 'skew'} str
+        """
+        def gaus(x):
+            return x * np.exp(-1 * x**2 / 2)
+
+        def skew(x):
+            return x**2
```

Comment:
According to the [FastICA documentation](http://scikit-learn.org/stable/modules/generated/sklearn.decomposition.FastICA.html), the cost functions need to return the derivative too.

Reply:
Do you think we should support passing a non-standard cost function ? This was not supported in `mdp`, although there were more cost functions available there....